File: Cpp/203_Remove_Linked_List_Elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        if(!head) return head;
        
        while(head && head -> val == val) {
            head = head -> next;
        }
        
        ListNode* node = head;
        while(node && node->next){
            if(node->next->val == val){
                ListNode* temp = node->next->next;
                node->next = temp;
            } else {
                node = node->next;
            }
        }
        
        return head;
    }
};

File: Kotlin/203_Remove_Linked_List_Elements.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun removeElements(head: ListNode?, `val`: Int): ListNode? {
        var node = head
        
        while(node?.next != null) {
            if(node.next?.`val` == `val`) {
                node.next = node.next?.next
            } else {
                node = node.next!!
            }
        }
        
        if(head?.`val` == `val`) return head.next
        
        return head
    }
}